nixos/modules/services/web-servers/nginx/tailscale-auth.nix

```diff
@@ -0,0 +1,173 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.nginx.tailscaleAuth;
+in
+{
+  options.services.nginx.tailscaleAuth = {
+    enable = mkEnableOption (lib.mdDoc "tailscale.nginx-auth");
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.tailscale-nginx-auth;
+      defaultText = literalExpression "pkgs.tailscale-nginx-auth";
+      description = lib.mdDoc ''
+        The package that provides tailscale.nginx-auth
+      '';
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "tailscale-nginx-auth";
+      description = lib.mdDoc "User which runs tailscale-nginx-auth";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "tailscale-nginx-auth";
+      description = lib.mdDoc "Group which runs tailscale-nginx-auth";
+    };
+
+    expectedTailnet = mkOption {
+      default = "";
+      type = types.nullOr types.str;
+      example = "tailnet012345.ts.net";
+      description = lib.mdDoc ''
+        If you want to prevent node sharing from allowing users to access services
+        across tailnets, declare your expected tailnets domain here.
+      '';
+    };
+
+    socketPath = mkOption {
```

Comment:
This is probably limited to /run/ by systemd.

Reply:
Is this intended to be a descriptive statement? Because I think it's true but I don't see how it hurts.

Reply:
yeah whatever, fine like this